config/apply/ccsm/translate.js

```diff
@@ -435,53 +424,47 @@ export function translateResponse(customApiResponse, patientData) {
       conclusionCodes = mapResults(excisionResults, customExcisionCodes, standardHistologyCodes, conclusionCodes);
     }
 
-    if (diagnosticReportIndex !== -1) {
-      console.log('Found the diagnostic report reference');
-      // Update the DiagnosticReport in patientData with the results from the custom API
-      patientData[diagnosticReportIndex] = {
-        ...patientData[diagnosticReportIndex],
-        code: {
-          coding: [
-            ...patientData[diagnosticReportIndex].code.coding,
-            ...codings
-          ]
-        },
-        conclusionCode: conclusionCodes
-      };
+    let patient = patientData.find(pd => pd.resourceType === 'Patient');
+
+    // Create a DiagnosticReport resource from Order
+    let newDiagnosticReport = {
+      'resourceType': 'DiagnosticReport',
+      'id': order.OrderId,
+      'subject': { 'reference': 'Patient/' + patient.id },
+      'code': { 'coding': codings },
+      'conclusionCode': conclusionCodes,
+      'effectiveDateTime': order.ResultDate,
```

Comment:
I think for consistency we should pull out the resultDate when we deconstruct the order object at the top of the logic in this file (around line 372 - 382,  beginning of `translateResponse()` function), so that `effectiveDateTime` can be taken from `resultDate` rather than by reaching back into the `order` object.

Reply:
It is not necessary to deconstruct the `order` object since all the element value could be retrieved using `order.{elementname}`. There are some benefit for doing that, such as renaming, and default value. Since order.ResultDate is onetime use here, the code does not take any advantages from deconstruct.

Reply:
Okay. Makes sense that since we're only using it once here there isn't really a need to retrieve it earlier-on in the function. I'll resolve this thread.